src/algebra/jordan/special.lean

```diff
@@ -0,0 +1,198 @@
+-/
+
+/--
+The symmetrised algebra has the same underlying space as the original algebra.
+-/
+def sym_alg (α : Type*) : Type* := α
+
+postfix `ˢʸᵐ`:std.prec.max_plus := sym_alg
+
+namespace sym_alg
+
+variables {α : Type*}
+
+/-- The element of `sym_alg α` that represents `a : α`. -/
+@[pp_nodot]
+def sym : α → αˢʸᵐ := id
+
+/-- The element of `α` represented by `x : αˢʸᵐ`. -/
+@[pp_nodot]
+def unsym : αˢʸᵐ → α := id
+
+instance [inhabited α] : inhabited αˢʸᵐ := ⟨sym (default α)⟩
+
+@[simp] lemma unsym_sym (x : α) : unsym (sym x) = x := rfl
+@[simp] lemma sym_unsym (x : α) : sym (unsym x) = x := rfl
+
+instance [has_zero α] : has_zero (αˢʸᵐ) := { zero := sym 0 }
+instance [has_sub α] : has_sub αˢʸᵐ := { sub := λ a b, sym (unsym a - unsym b) }
+
+instance [has_add α] : has_add αˢʸᵐ :=
+{ add := λ a b, sym (unsym a + unsym b) }
+
+@[simp] lemma sym_add [has_add α] (a b : α) : sym (a + b) = sym a + sym b := rfl
+@[simp] lemma unsym_add [has_add α] (x y : αˢʸᵐ) : unsym (x + y) = unsym x + unsym y := rfl
+
+@[simp] lemma sym_sub [has_sub α] (x y : α) : sym (x - y) = sym x - sym y := rfl
+@[simp] lemma unsym_sub [has_sub α] (x y : αᵐᵒᵖ) : unsym (x - y) = unsym x - unsym y := rfl
+
+instance [has_neg α] : has_neg αˢʸᵐ :=
+{ neg := λ a, sym (-unsym a) }
+
+instance {R : Type*} [has_scalar R α] : has_scalar R αˢʸᵐ :=
+{ smul := λ r a, sym (r • unsym a) }
+
+@[simp] lemma sym_smul {R : Type*} [has_scalar R α] (c : R) (a : α) : sym (c • a) = c • sym a := rfl
+
+lemma unsym_injective : function.injective (@unsym α) := λ _ _, id
+
+instance [add_comm_semigroup α] : add_comm_semigroup (αˢʸᵐ) :=
+unsym_injective.add_comm_semigroup _ (λ _ _, rfl)
+
+instance [add_monoid α] : add_monoid (αˢʸᵐ) :=
+unsym_injective.add_monoid_smul _ rfl (λ _ _, rfl) (λ _ _, rfl)
+
+instance [add_group α] : add_group (αˢʸᵐ) :=
+unsym_injective.add_group_smul _ rfl
+  (λ _ _, rfl) (λ _, rfl) (λ _ _, rfl) (λ _ _, rfl) (λ _ _, rfl)
+
+instance [add_comm_monoid α] : add_comm_monoid (αˢʸᵐ) :=
+{ ..sym_alg.add_comm_semigroup, ..sym_alg.add_monoid }
+
+instance [add_comm_group α] : add_comm_group (αˢʸᵐ) :=
+{ ..sym_alg.add_comm_monoid, ..sym_alg.add_group }
+
+
+
+@[simp] lemma sym_zero [has_zero α] : sym (0 : α) = 0 := rfl
+@[simp] lemma unsym_zero [has_zero α] : unsym (0 : αˢʸᵐ) = 0 := rfl
+
+@[simp] lemma unsym_eq_zero_iff {α} [has_zero α] (a : αˢʸᵐ) : a.unsym = (0 : α) ↔ a = (0 : αˢʸᵐ) :=
+unsym_injective.eq_iff' rfl
+
+instance {R : Type*} [semiring R] [add_comm_monoid α] [module R α] : module R αˢʸᵐ :=
+function.injective.module R ⟨unsym, rfl, λ _ _, rfl⟩ (λ _ _, id) (λ _ _, rfl)
+
+/- Introduce the symmetrised multiplication-/
+instance [has_add α] [has_mul α] [has_one α] [invertible (2 : α)] : has_mul(αˢʸᵐ) :=
+{ mul := λ a b, sym (⅟2 * (unsym a * unsym b + unsym b * unsym a)) }
+
+lemma mul_def [ring α] [invertible (2 : α)] (a b : αˢʸᵐ) :
+  a * b = sym (⅟2*(unsym a * unsym b + unsym b * unsym a)) := by refl
```

Comment:
Can you add the lemmas
```
unsym_mul (a b : αˢʸᵐ) : unsym (a * b) = ⅟2*(unsym a * unsym b + unsym b * unsym a)
sym_mul_sym (a b : α) : sym a * sym b = ⅟2*(a * b + b * a)
```

Reply:
For the second equation, do you mean:
```
sym_mul_sym (a b : α) : sym a * sym b = sym (⅟2*(a * b + b * a))
```
?

Reply:
I've added `unsym_mul` and `sym_mul_sym`.